assets/app/scripts/controllers/route.js

```diff
@@ -62,6 +63,11 @@ angular.module('openshiftConsole')
             };
           });
 
+        // List services to display route warnings.
+        DataService.list("services", context, function(services) {
```

Comment:
I guess we could remove the popover entirely if all warnings are cleared, but the hover stops working if you change the message.